src/simplewallet/simplewallet.cpp

```diff
@@ -362,6 +362,29 @@ bool simple_wallet::seed_set_language(const std::vector<std::string> &args/* = s
   return true;
 }
 
+bool simple_wallet::change_password(const std::vector<std::string> &args)
+{
+  // watch-only wallets don't need passwords, right?
+  if(m_wallet->watch_only())
+  {
+    fail_msg_writer() << tr("wallet is watch-only and has no seed");
+    return true;
+  }
+  
+  // prompts for a new password, this is not a new wallet so pass in false.
+  const auto pwd_container = tools::wallet2::password_prompt(false);
+  
+  if(!pwd_container)
+  {
+    fail_msg_writer() << tr("no password given");
+    return true;
+  }
+  
+  m_wallet->rewrite(m_wallet_file, pwd_container->password());
```

Comment:
This new function returns bool, and `wallet2::rewrite` throws exceptions on errors. The expectation here I think is to catch the type of exceptions `rewrite` is throwing, log the error, then return false. Propagating the error further (i.e. let exception pass-through) doesn't seem like it would be useful in this situation.